experimental/dds/tree2/src/feature-libraries/chunked-forest/codec/chunkEncodingGeneric.ts

```diff
@@ -100,15 +120,37 @@ export function handleShapesAndIdentifiers<TEncodedShape>(
 	};
 }
 
+/**
+ * A tree shape which can can references to it deduplicated using {@link Counter}.
+ *
+ * @remarks
+ * Deduplication of shapes is done after the output is otherwise generated (including identifier dictionary encoding).
+ * To avoid having to decode the decode the data array to determine which data is a shape and which is some other object,
+ * some recognizable representing is required.
+ * Using a class and checking its prototype works for this, and is why Shape is a class.
+ *
+ * Note that deduplication compares shapes by object identity not by content, so encoders must ensure shapes are not duplicated to achieve efficient encoding.
+ */
 export abstract class Shape<TEncodedShape> {
 	/**
 	 * Count this shape's contents.
+	 *
+	 * Used to discover referenced shapes (which need to be included),
```

Comment:
The parenthetical there is not about getting an accurate count (though that also matters), but instead about ensuring the more important property that referenced shapes be included in the serialized shape list at all (so you don't crash in encodeShape trying to make a reference to a shape that is not included because nothing claimed to reference it).

I'll update the docs to clarify.